### ISSUE TITLE
`foreign` function analysis?

### ISSUE THREAD

**Issue by WeeknightMVP** [GaloisInc/cryptol#1574]
Can the Remote API or another tool identify ...?
- modules with `foreign` functions but no shared library
- `foreign` functions that...
  + have no implementation in a shared library
  + have no fallback (specification)

`:browse` in the REPL and `focused_module` and `names` methods of `CryptolConnection` in the Remote API do not report these.  `:file-deps` and `file_deps` respectively report a `foreign` dependency when a shared library is present, but when absent this field is empty, as it would be for a module with no `foreign` functions.

**Comment by qsctr** [comment]
I don't think this information is currently exposed anywhere (other than the warning messages shown when loading a module with missing foreign implementations) but it would not be too hard to add it.

**Comment by weaversa** [comment]
It would also be helpful to have line numbers.
